### PR TITLE
feat: add client-core metrics to veneer 

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java [hunk 1]
```diff
@@ -311,6 +312,15 @@ private void configureMetricsBridge(Builder settings) {
     MetricsApiTracerAdapterFactory metricsApiTracerAdapterFactory =
         new MetricsApiTracerAdapterFactory();
     settings.stubSettings().setTracerFactory(metricsApiTracerAdapterFactory);
+    // Set up channel interceptor to record channels.channel$id.rpc.latency metrics
+    if (settings.stubSettings().getTransportChannelProvider()
+        instanceof InstantiatingGrpcChannelProvider) {
+      InstantiatingGrpcChannelProvider.Builder channelProvider =
+          ((InstantiatingGrpcChannelProvider) settings.stubSettings().getTransportChannelProvider())
+              .toBuilder();
+      channelProvider.setInterceptorProvider(new MetricsChannelInterceptorInterceptorProvider());
```
Comment (mutianf) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java:321]: Note: Not sure about this part.. this won't include the interceptors that's already set in the setting, but looks like there isn't a way in InstantiatingGrpcChannelProvider to get interceptor provider from the builder https://github.com/googleapis/gax-java/blob/master/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java#L468-L471?

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/metrics/MetricsApiTracerAdapter.java [hunk 2]
```diff
@@ -29,16 +37,31 @@
 @InternalApi
 public class MetricsApiTracerAdapter implements ApiTracer {
 
+  private final Timer firstResponseLatencyTimer =
```
[the remark this answers is not in the document]
Comment (mutianf) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/metrics/MetricsApiTracerAdapter.java:40]: I think this metric won't show up in metric registery https://github.com/googleapis/java-bigtable-hbase/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResponseQueueReader.java#L87-L106 (line 94 is where it's recorded)

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableVeneerApi.java [hunk 3]
```diff
@@ -63,5 +84,18 @@ public DataClientWrapper getDataClient() {
   public void close() throws IOException {
     dataClientWrapper.close();
     adminClientWrapper.close();
+    activeSessions.dec();
+    for (int i = 0; i < channelPoolSize; i++) {
+      BigtableClientMetrics.counter(MetricLevel.Info, "grpc.channel.active").dec();
+    }
```
Comment (igorbernstein2) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableVeneerApi.java:90]: Doesnt this need to account for channel pooling as well? ie defer decrementing to the factory?

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]
looks good, but needs to fix channel dec when channelpooling is enabled

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableVeneerApi.java (hunk 3 above)
Comment (igorbernstein2) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableVeneerApi.java:90]: actually nevermind, channelPoolSize will be 0

**Review by igorbernstein2 — APPROVED** [review]
lgtm

**Comment by igorbernstein2** [comment]
closing in favor of #2978 (which fixes a minor dep issue)

**Pull request closed without merging**